crates/solver/src/solver.rs

```diff
@@ -171,24 +172,29 @@ pub fn create(
     ));
     let http_solver_cache = http_solver::InstanceCache::default();
     // Helper function to create http solver instances.
-    let create_http_solver =
-        |account: Account, url: Url, name: &'static str, config: SolverConfig| -> HttpSolver {
-            HttpSolver::new(
-                DefaultHttpSolverApi {
-                    name,
-                    network_name: network_id.clone(),
-                    chain_id,
-                    base: url,
-                    client: client.clone(),
-                    config,
-                },
-                account,
-                native_token,
-                token_info_fetcher.clone(),
-                buffer_retriever.clone(),
-                http_solver_cache.clone(),
-            )
-        };
+    let create_http_solver = |account: Account,
+                              url: Url,
+                              name: &'static str,
+                              config: SolverConfig,
+                              use_internal_buffers: bool|
```

Comment:
Since this gets passed to the `DefaultHttpSolverApi` anyway, can't this be included in the config `SolverConfig` above?

This might make it require way less plumbing.

Reply:
The SolverConfig is common to all solvers, so we would have two parameter there, one called `use_internal_buffers` and another `has_use_internal_buffers_parameter`. What about if we replace both by an Option<bool> ?

Reply:
Right now, we assume that our HTTP solvers have the same interface. We can make it an option so that:
```
None        => ?...
Some(false) => ?...&use_internal_buffers=false&...
Some(true)  => ?...&use_internal_buffers=true&...
```

Reply:
So the issue is indeed that http solvers do not have the same interface. Quasimodo is the only one that supports this parameter. The idea regarding the Option is to offer the 3 states we need: no parameter, parameter=false, parameter=true. Is this what you're saying?